OpenKh.Patcher/PatcherProcessor.cs

```diff
@@ -279,5 +284,79 @@ private static void PatchAreaDataScript(Context context, List<AssetFile> sources
 
             Kh2.Ard.AreaDataScript.Write(stream.SetPosition(0), scripts.Values);
         }
+
+        private static void PatchListReplace(Context context, List<AssetFile> sources, Stream stream)
+        {
+            foreach (var source in sources)
+            {
+                var deserializer = new DeserializerBuilder().IgnoreUnmatchedProperties().Build();
+                Dictionary<string, byte> characterMap = new Dictionary<string, byte>() {
+                            { "Sora", 1 }, { "Donald", 2 }, { "Goofy", 3 },  { "Mickey", 4 },  { "Auron", 5 }, { "PingMulan",6 }, { "Aladdin", 7 },  { "Sparrow", 8 }, { "Beast", 9 },  { "Jack", 10 },  { "Simba", 11 }, { "Tron", 12 }, { "Riku", 13 }, { "Roxas", 14}
+                        };
+                    string sourceText = File.ReadAllText(context.GetSourceModAssetPath(source.Name));
+                    if (source.Name.Contains("enc"))
+                    {
+                        byte[] data = System.Convert.FromBase64String(sourceText);
+                        sourceText = System.Text.ASCIIEncoding.ASCII.GetString(data);
+                    }
+                    switch (source.Type)
+                    {
+                        case "trsr":
+                            var TrsrList = Kh2.SystemData.Trsr.Read(stream).ToDictionary(x => x.Id, x => x);
+                            var moddedTrsr = deserializer.Deserialize<Dictionary<int, Kh2.SystemData.Trsr>>(sourceText);
+                            foreach (KeyValuePair<int, Kh2.SystemData.Trsr> treasure in moddedTrsr)
+                            {
+                                TrsrList[treasure.Value.Id].ItemId = treasure.Value.ItemId;
+                            }
+                                Kh2.SystemData.Trsr.Write(stream.SetPosition(0), TrsrList.Values);
+                            break;
+
+                        case "item":
+                            var ItemList = Kh2.SystemData.Item.Read(stream);
+                            var moddedItem = deserializer.Deserialize<Kh2.SystemData.Item>(sourceText);
+                            ItemList = moddedItem;
+                            ItemList.Write(stream.SetPosition(0));
+                            break;
+
+                        case "fmlv":
+                            var FormList = Kh2.Battle.Fmlv.Read(stream).ToDictionary(x => String.Concat(x.FormFm, x.FormLevel), x => x);
```

Comment:
I would suggest to create a Dictionary where the key is the `FormFm` and the value is a list of `Kh2.Battle.Fmlv.Level`, where every index of the list represents a level. This would help to potentially add new levels to individual drive forms in the future. Although this might require a change to your YAML. In my mind it would become something like the following:
```yml
Wisdom:
  - FormLevel: 2
    Ability: 3
    Exp: 1234
```
where you can take the "Wisdom" name out of it to know in which index of the original file you have to patch the values. It would be also important to allow the same files to patch both Vanilla and the FinalMix versions of the game. Although I realise it might be complex.

There is also an update on how the `Fmlv` is parsed https://github.com/Xeeynamo/OpenKh/blob/f6471f0339f0a625ad69f0b4700efe50ae24f36a/OpenKh.Kh2/Battle/Fmlv.cs . I believe this is a complex problem to solve at the moment. Finally, to prevent to break the compatibility with future renaming to the parser, could you create a dedicated class that maps to the actual `Fmlv`?